### PR TITLE
XWayland: reduce tight coupling between surface components

### PR DESCRIPTION
**Pull request by wmww**
Adds two new interfaces (`XWaylandSurfaceRoleSurface` and `XWaylandSurfaceObserverSurface`) which are used by their respective components to talk to the XWayland surface. There's pleanty more work to do in this area, but this is a start.

### CODE REVIEW

**Comment by wmww** [comment]
bors try

**Comment by wmww** [comment]
Ah! CI is blowing up because I forgot to add a file

File: src/server/frontend_xwayland/xwayland_surface.cpp [hunk 1]
```diff
@@ -98,12 +98,17 @@ mf::XWaylandWMSurface::XWaylandWMSurface(
...
-mf::XWaylandWMSurface::~XWaylandWMSurface()
+mf::XWaylandSurface::~XWaylandSurface()
 {
     close();
 }
 
-void mf::XWaylandWMSurface::close()
+void mf::XWaylandSurface::map()
+{
+    scene_surface_state_set(mir_window_state_restored); // TODO: use the real window state
```
Comment (AlanGriffiths) [inline src/server/frontend_xwayland/xwayland_surface.cpp]: Is the TODO intended to land?

File: src/server/frontend_xwayland/xwayland_surface_observer_surface.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+{
+namespace frontend
+{
+/// The interface used by XWaylandSurfaceObserver to talk to the surface object
+class XWaylandSurfaceObserverSurface
+{
+public:
+    virtual void scene_surface_state_set(MirWindowState new_state) = 0;
+    virtual void scene_surface_resized(const geometry::Size& new_size) = 0;
+    virtual void scene_surface_close_requested() = 0;
```
Comment (AlanGriffiths) [inline src/server/frontend_xwayland/xwayland_surface_observer_surface.h]: This naming style is a bit clumsy, but at the separation makes sense.

File: src/server/frontend_xwayland/xwayland_surface.cpp (hunk 1 above)
Comment (wmww) [reply src/server/frontend_xwayland/xwayland_surface.cpp]: Yes, for now

**Comment by wmww** [comment]
bors r=AlanGriffiths

**Pull request merged**